imports/plugins/core/orders/client/containers/orderDashboardContainer.js

```diff
@@ -194,6 +260,397 @@ class OrderDashboardContainer extends Component {
+          ordersToRegress++;
+        } else if (!order.shipping[0].workflow.workflow.includes("coreOrderWorkflow/picked") &&
+        (order.shipping[0].workflow.status === "coreOrderWorkflow/packed" ||
+        order.shipping[0].workflow.status === "coreOrderWorkflow/labeled" ||
+        order.shipping[0].workflow.status === "coreOrderWorkflow/shipped")) {
+          ordersToRegress++;
+        }
+      }
+    });
+
+    // display regression alert if order(s) are being regressed
+    if (ordersToRegress) {
+      this.displayRegressionAlert(selectedOrders, ordersToRegress, status);
+    } else {
+      // set status to 'picked' if order(s) are in the previous state
+      if (isNotPicked) {
+        this.shippingStatusUpdateCall(selectedOrders, status);
+
+        // display alert if order(s) are already in this state
+      } else if (!isNotPicked && isPicked) {
+        Alerts.alert({
+          text: i18next.t("order.orderAlreadyInState", {
+            orderText: this.displayOrderText(selectedOrders), status: status
+          })
+        });
+      }
+    }
+  }
+
+  packedShippingStatus = (selectedOrders, status) => {
+    let isNotPicked = 0;
+    let isNotPacked = 0;
+    let isPacked = 0;
+    let ordersToRegress = 0;
+    const whichFalseState = shippingStates.picked;
+
+    selectedOrders.forEach((order) => {
+      // check if the order(s) are in this state already or in one of the previous states
+      if (order.shipping[0].workflow.status === "new") {
+        isNotPicked++;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/picked") {
+        isNotPacked++;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/packed") {
+        isPacked++;
+      } else {
+        // check if the selected order(s) are being regressed back to this state
+        if (order.shipping[0].workflow.workflow.includes("coreOrderWorkflow/packed")) {
+          ordersToRegress++;
+        } else if (!order.shipping[0].workflow.workflow.includes("coreOrderWorkflow/packed") &&
+          (order.shipping[0].workflow.status === "coreOrderWorkflow/labeled" ||
+          order.shipping[0].workflow.status === "coreOrderWorkflow/shipped")) {
+          ordersToRegress++;
+        }
+      }
+    });
+
+    // display regression alert if order(s) are being regressed
+    if (ordersToRegress) {
+      this.displayRegressionAlert(
+        selectedOrders, ordersToRegress, status, whichFalseState, isNotPicked, isNotPacked, isPacked
+      );
+
+      // display proper alert if the order(s) are in this state already or want to skip the previous states
+    } else this.displayAlert(selectedOrders, status, whichFalseState, isNotPicked, isNotPacked, isPacked);
+  }
+
+  labeledShippingStatus = (selectedOrders, status) => {
+    let isNotPacked = 0;
+    let isNotLabeled = 0;
+    let isLabeled = 0;
+    let ordersToRegress = 0;
+    let whichFalseState = "";
+
+    selectedOrders.forEach((order) => {
```

Comment:
I've seen this bit of code 3 times now. Would like to have some comments and documentation explaining what the difference is between these very similar functions.

All the comments I've made on previous ones apply here too.

Additionally, worth adding a `TODO` task to write some helper functionality that can reduce the repetition and increase readability.